src/main/kotlin/org/rust/ide/template/postfix/stringBasedPostfixTemplates.kt

```diff
@@ -61,3 +68,37 @@ class MatchPostfixTemplate(provider: RsPostfixTemplateProvider) :
         }
     }
 }
+
+abstract class ForPostfixTemplate(provider: RsPostfixTemplateProvider, pred: (RsExpr) -> Boolean, example: String) :
+    StringBasedPostfixTemplate("iter", example, RsTopMostInScopeSelector(pred), provider) {
+
+    override fun getElementToRemove(expr: PsiElement): PsiElement = expr
+}
+
+
+private val RsExpr.implementsIntoIter: Boolean
+    get() = isImplementsTrait(this, "iter::IntoIterator")
+
+private val RsExpr.implementsIter: Boolean
+    get() = isImplementsTrait(this, "iter::Iterator")
+
+fun isImplementsTrait(expr: RsExpr, traitName: String, vararg subst: Ty): Boolean {
+    val items = StdKnownItems.relativeTo(expr)
+    val implLookup = ImplLookup(expr.project, items)
+    val trait = items.findCoreItem(traitName) as? RsTraitItem ?: return false
+    return implLookup.canSelect(TraitRef(expr.type, trait.withSubst(*subst)))
+}
+
+class IterPostfixTemplate(provider: RsPostfixTemplateProvider)
+    : ForPostfixTemplate(provider, RsExpr::implementsIter, "for x in expr") {
+
+    override fun getTemplateString(element: PsiElement): String =
+        "for x in ${element.text} {\n\$END$\n}"
+}
+
+class IntoIterPostfixTemplate(provider: RsPostfixTemplateProvider)
+    : ForPostfixTemplate(provider, RsExpr::implementsIntoIter, "for x in expr.into_iter()") {
+
+    override fun getTemplateString(element: PsiElement): String =
+        "for x in ${element.text}.into_iter() {\n\$END$\n}"
```

Comment:
Why do we need this explicit call of `into_iter` method?
Rust can call it [implicitly](https://doc.rust-lang.org/std/iter/index.html#for-loops-and-intoiterator)